### ISSUE TITLE
Excavator Mineral Veins Should Match Consumption

### ISSUE THREAD

**Issue by voidsong-dragonfly** [BluSunrize/ImmersiveEngineering#5891]
### Explain your idea

Currently, the Excavator mineral veins don't often match the consumption of materials that IE especially, but IE and other mods have. This makes the excavator significantly more hassle to use than it should be and produces a lot of 'useless' items and blocks that sit around in silos. This conversation was had in the discord with @Shadows-of-Fire and a few others.

The excavator currently has nickel and sulfur in the most excess from common veins, but materials can be broadly grouped into several categories, each ranked roughly in importance to IE:

#### 1. Common Ores:

1. Iron - Iron is used in everything in massive quantity in IE.
2. Coal - Steel is by far the largest consumption in late IE and for steel you need one coal per.
3. Copper - Copper use falls off in late IE but it's still an essential component.
4. Redstone - Redstone is used for control, engineering blocks, and more.

#### 2. Specialty Ores:

1. Aluminium - I can't decide if aluminium is specialty or mainstay. It's used in lategame infrastructure all over, so consider it a halfway point between common and specialty ores.
2. Gold, Silver - The use of these is basically coupled in IE because everything that uses one, uses electrum. 
3. Nickel - Nickel is used for thermos, bullets, and for the diesel generator. That's about it.
4. Lead - Lead is used for bullets and the lowest tier of accumulator, plus dyes.
5. Uranium - It makes power in block form. Looks green.
6. Sulfur - Sulfur is used in bullets, and that is about it. Maybe it could be added to make more fertilizer?
7. Lapis Lazuli - Unused in IE because it's not needed, but otherwise useful in various ways.

#### 3. Rare Ores:

1. Emerald - IE adds some very good villager trades, so emeralds are quite useful to have.
2. Diamond - Diamond is used in Vanilla and in sawblades, and Vanilla by far consumes more.
3. Ancient Debris - No uses for Netherite in IE directly, but it's used in Vanilla in quantity

Due to the issues people have with procuring aluminium, the veins it makes the most sense to focus on are thus _Iron, Coal, Redstone, Copper,_ and _Aluminium_. As other resources are needed, they should have smaller veins and maybe be recoverable from 'side' veins that are significantly rarer.

Sulfur is currently gotten in massive quantity from ore veins and it's barely used, which is one of the main things to fix. Iron ore is not a supermajority in _any vein_, which makes getting iron ore annoying. Thus, I suggest the following changes:

 - [ ] Laterite should be bauxite/iron/nickel in 60/30/10. IRL laterites can be nickel and aluminium bearing along with iron, and it provides a source of nickel without going through the next change for massive quantities:
 - [ ] The platinum vein should be scrapped and folded into Pentlandite, as 50/20/10/10/10 nickel/iron/sulfur/platinum/osmium. This keeps a mostly-nickel vein around while also providing a vein with both osmium and platinum as necessary.
 - [ ] There should be some ≥80% iron ore vein so industrial quantities of iron can be harnessed, similar to Bituminous Coal. I suggest Banded Iron, a roughly 80/20 mixture of iron/dripstone, with the dripstone substituting for sedimentary chert.
 - [ ] Auricupride could (this I am iffy on) be converted to a copper/gold/silver vein. Gold and silver can sometimes be found together, as sometimes can gold and copper and silver and copper. This would make Auricupride a 'wiring' deposit and provide gold and copper together. This suggestion is unneeded and Auricupride can stay how it was if the below happens, and I would suggest the second option as gold is used a fair bit in Vanilla (piglin trading, etc):
 - [ ] Galena should get some gold as well, as it often occurs with galena. I suggest a 40/25/25/10 lead/silver/gold/sulfur, which is massively high for the precious metals but also makes Galena a perfect ore for wiring without having sulfur loads be egregious.
 - [ ] Chalcopyrite should have sulfur go down, copper/iron go up. I don't know whether to even keep chalcopyrite at all with the new Banded Iron vein, but it's a decent 'all-rounder' so I think it would be reasonable to stay. Maybe 55/35/10 iron/copper/sulfur to make it a significantly better vein for 'basic extraction'
 - [ ] Lazulitic Intrusion should be added as a vein for lapis. The name is completely made up, but lapis isn't often found in a larger mineral, it _is_ the larger mineral. So. 75/15/10 lapis/gold/sulfur i think, gold isn't found with lapis IRL - it's pyrite - but if you're doing enchanting ( a reason to use lapis) or Ars Nouveau (another) you're probably also using gold for various magic reasons, and it fits with the looks of lapis better than the pyritic iron/sulfur mix.

Redstone and coal veins are basically currently fine, I don't have major issues with their distributions. Vein weights in worldgen may need to be rebalanced, with metals balanced up and stones/decorative blocks balanced down. As well, it was suggested in the Discord to make the excavator load a few chunks around itself so there aren't production stoppages.

Shadow also suggested in the discord to make the Wolframite either compat or otherwise not nearly so tungsten heavy as the mods with tungsten (which ones?) don't use a lot of it. I don't know how best to handle that.

**Comment by BluSunrize** [comment]
Adding thoughts of my own to this:

- Both the lapis and redstone veins _could_ include calcite? It's a thing found alongside lapis, though I guess it doesn't have enough usages within IE or vanilla to warrant this?
- Veins for tin and tungsten could potentially be combined, since cassiterite and wolframite do appear together?
- You didn't actually suggest anything for Diamond and I'm struggling to see how we'd do that one. Kimberlite is basically the only irl mineral for it (and it's a cool name!) but I don't really know what other things to put in there...

**Comment by voidsong-dragonfly** [comment]
A way you could do diamond is some sort of alluvial placer deposit, as that's where a lot of diamonds were found IRL, after erosion washed them out of their kimberlite pipes. It would essentially be rarer silt vein with some other changes to make it more of a specifically riverine deposit and not generic silt, would be how I would go about it.

Lapis and redstone including calcite: I mean, _sure_, but most of the veins are ore-only? How much do you want to break that precedent by adding stones too? We already have a 'White Cliffs of Dover' vein that I added last time I added new veins. (see below [here](https://github.com/BluSunrize/ImmersiveEngineering/blob/f0d26f797d9fa516f0ed7f39e570f08d93a1a903/src/datagen/java/blusunrize/immersiveengineering/data/Recipes.java#L1334) from 1.20.1)
```java
MineralMixBuilder.builder(overworld)
    .addSeabedSpoils()
    .addOre(Blocks.CALCITE, .65f)
    .addOre(Blocks.DRIPSTONE_BLOCK, .3f)
    .addOre(Blocks.BONE_BLOCK, .05f)
    .setWeight(15)
    .setFailchance(.05f)
    .build(out, toRL("mineral/ancient_seabed"));
```

**Comment by BluSunrize** [comment]
Closed by #5902
